pipelines/tasks/get-package-versions-list.yaml

```
@@ -0,0 +1,36 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: get-package-versions-list
+spec:
+  volumes:
+    - name: unpinned-packages
+      emptyDir:
+        medium: memory
+    - name: script
+      configMap:
+        name: get-packages-script
+        defaultMode: 755
+  workspaces:
+    - name: pinned-packages
+  params:
+    - name: PACKAGES
+      type: array
+  stepTemplate:
+    volumeMounts:
+      - name: unpinned-packages
+        mountPath: /scratchspace
+      - name: script
+        mountPath: /scripts
+  steps:
+    - image: registry.access.redhat.com/ubi9-micro
+      args:
+        - /scripts/get-packages
+        - /scratchspace/requirements.in
+        - $(params.PACKAGES) # We need to use a script because of how array parameters expansion works in Tekton ( https://github.com/tektoncd/pipeline/blob/main/docs/tasks.md#substituting-array-parameters )
```

hmm... this is not being executed I'm afraid.

The next step fails with:

```
Error: Invalid value for '[SRC_FILES]...': Path '/scratchspace/requirements.in' does not exist.
```

---

That looks like the consequence of the above, or do you mean with those fixed

---

right, even with the other fixes this step is failing to execute. It works for me when I point to the shell as the command to run:

```suggestion
        - $(params.PACKAGES) # We need to use a script because of how array parameters expansion works in Tekton ( https://github.com/tektoncd/pipeline/blob/main/docs/tasks.md#substituting-array-parameters )
      command: [sh]
      name: extract-requirements
```

I would also suggest to `name` the steps (e.g. `extract-requirements`, `pip-compile`)

---

Ah, I think I get what's going on, I'll use the script as the `command`. Otherwise (using sh) I'm not sure what happend with args